includes/abstracts/llms-abstract-user-engagement.php

```diff
@@ -200,18 +200,32 @@ public function is_awarded() {
 	 * Allow child classes to merge the post content based on content from the template.
 	 *
 	 * @since 6.0.0
+	 * @since [version] Added optional `$content` and `$load_reusable_blocks` parameters.
 	 *
+	 * @param string $content              Optionally use the given content instead of `$this->content`.
+	 * @param bool   $load_reusable_blocks Optionally replace reusable blocks with their actual blocks.
 	 * @return string
 	 */
-	public function merge_content() {
+	public function merge_content( $content = null, $load_reusable_blocks = false ) {
 
-		return $this->get( 'content', true );
+		if ( is_null( $content ) ) {
+			$content = $this->get( 'content', true );
+		}
+
+		if ( $load_reusable_blocks ) {
+			$blocks  = parse_blocks( $content );
+			$blocks  = LLMS_Forms::instance()->load_reusable_blocks( $blocks );
```

Comment:
This line is the only coding problem I see in the entire PR... It works but now that we're using this outside of forms I'm feeling like the content of the `load_reuseable_blocks()` method could (should?) be moved to it's own utility function and then we can call that function here (and the forms class can call it as well). We should probably then deprecate the forms method. This does "inflate" the size of this PR... It's probably better that we just open up an enhancement issue to track this and do it at some point in the future.

Reply:
> I did find what could be kind of a problem though...

Thank you for catching this! I've committed a fix for this, but have not written a JavaScript test.

I've never been able to get the JavaScript testing framework working in my dev environment. It's been one npm package problem after another. Maybe this is the time to work through all the issues. Ugh.

> It works but now that we're using this outside of forms I'm feeling like the content of the load_reuseable_blocks() method could (should?) be moved to it's own utility function and then we can call that function here (and the forms class can call it as well). We should probably then deprecate the forms method.

I agree completely. I'll create a new issue for that.